### PR TITLE
refactor: remove int inheritance from Version

### PR DESCRIPTION
**Pull request by jjaakola-aiven**
# About this change - What it does

Make Version a container class for integer for stronger typing.

### CODE REVIEW

File: karapace/typing.py [hunk 1]
```diff
@@ -59,23 +59,47 @@ class Mode(StrEnum):
...
+        return self._value >= other.value
+
+    def __lt__(self, other: Version) -> bool:
+        return self._value < other.value
+
+    def __le__(self, other: Version) -> bool:
+        return self._value <= other.value
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, Version):
```
Comment (nosahama) [inline karapace/typing.py:92]: **[non-blocking]**
Would be nice to also have this check on the other comparison methods.

**Review by nosahama — DISMISSED** [review]

File: karapace/typing.py (hunk 1 above)
Comment (jjaakola-aiven) [reply karapace/typing.py:100]: Done.

File: karapace/typing.py [hunk 2]
```diff
@@ -59,23 +59,55 @@ class Mode(StrEnum):
...
+        return f"{int(self._value)}"
 
     def __repr__(self) -> str:
-        return f"Version={int(self)}"
+        return f"Version({int(self._value)})"
+
+    def __gt__(self, other: Version) -> bool:
+        if isinstance(other, Version):
+            return self._value > other.value
+        raise NotImplemented(f">Cannot compare {repr(other)} with {repr(self)}")
```
Comment (aiven-anton) [inline karapace/typing.py:82]: issue: `NotImplemented` is not an exception, and it should not be raised, it should be returned.

File: karapace/typing.py [hunk 3]
```diff
@@ -59,23 +59,55 @@ class Mode(StrEnum):
...
 
     def __str__(self) -> str:
-        return f"{int(self)}"
+        return f"{int(self._value)}"
 
     def __repr__(self) -> str:
-        return f"Version={int(self)}"
+        return f"Version({int(self._value)})"
+
+    def __gt__(self, other: Version) -> bool:
```
Comment (aiven-anton) [inline karapace/typing.py:79]: Usually you do not need to implement all of these methods, you can just implement one of them and decorate the class with `@functools.total_ordering`.

File: karapace/typing.py (hunk 1 above)
Comment (keejon) [reply karapace/typing.py:100]: @jjaakola-aiven think you have to either raise NotImplementedError or return NotImplemented

Comment (keejon) [reply karapace/typing.py:100]: ah Anton was faster :-)

File: karapace/typing.py (hunk 2 above)
Comment (aiven-anton) [reply karapace/typing.py:82]: It also isn't callable, these lines should be just simply `return NotImplemented`.

```pycon
>>> NotImplemented("Foo")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'NotImplementedType' object is not callable
```

File: karapace/typing.py (hunk 3 above)
Comment (nosahama) [reply karapace/typing.py]: `@functools.total_ordering` comes with some performance overhead, docs advise to implement these functions if the performance hit is too much.

**Review by nosahama — APPROVED** [review]

**Review by keejon — APPROVED** [review]

**Pull request merged**